src/OpenTelemetry.Instrumentation.Http/.publicApi/PublicAPI.Shipped.txt

```diff
@@ -0,0 +1,24 @@
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithException.get -> System.Action<System.Diagnostics.Activity, System.Exception>
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithException.set -> void
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithHttpRequestMessage.get -> System.Action<System.Diagnostics.Activity, System.Net.Http.HttpRequestMessage>
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithHttpRequestMessage.set -> void
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithHttpResponseMessage.get -> System.Action<System.Diagnostics.Activity, System.Net.Http.HttpResponseMessage>
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithHttpResponseMessage.set -> void
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithHttpWebRequest.get -> System.Action<System.Diagnostics.Activity, System.Net.HttpWebRequest>
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithHttpWebRequest.set -> void
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithHttpWebResponse.get -> System.Action<System.Diagnostics.Activity, System.Net.HttpWebResponse>
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.EnrichWithHttpWebResponse.set -> void
+OpenTelemetry.Instrumentation.Http.HttpClientTraceInstrumentationOptions.FilterHttpRequestMessage.get -> System.Func<System.Net.Http.HttpRequestMessage, bool>
```

Comment:
@vishweshbankwar @alanwest @CodeBlanch Should we rename the Filter APIs to use the term "With" similar to how we have named the Enrich APIs?

Reply:
"FilterWith" sounds odd to me. Probably use "FilterBy". But how does rename help here?

Reply:
> But how does rename help here?

The name should convey the intent of the API. `FilterHttpRequestMessage` sounds like an API that would filter the request message when in fact it merely uses the request message to filter out activities.

Comparing this to the Enrich APIs, how would `EnrichHttpRequestMessage` sound? We used "With" there to convey that the API would provide access to `HttpRequestMessage` to enrich the activity.

Reply:
I think we could rename it to something like `FilterUsingHttpRequestMessage` to make the intent more clear.

Reply:
Agree the naming could be improved, but I suggest its better to leave it as-is to avoid another breaking change at a late stage.

Reply:
I think it depends on how one interprets it. `FilterHttpRequestMessage` implies we are filtering the data collection. Filtering HttpRequestMessage has no other meaning as per my understanding. 

`EnrichHttpRequestMessage` --> This would be confusing, because you can actually enrich the request object (like adding additional headers etc.). So `With` makes it more clear IMO. 

But I don't have strong take on it except I feel "FilterBy" is more apt if we want to rename.

Reply:
> I suggest its better to leave it as-is to avoid another breaking change at a late stage.

Yeah true ☹️

Reply:
I also agree with the sentiment that the naming is not great, but it is not worth changing at this point in time.